cuda_bindings/cuda/bindings/_path_finder/load_nvidia_dynamic_library.py

```diff
@@ -38,6 +59,15 @@ def _load_nvidia_dynamic_library_no_cache(libname: str) -> LoadedDL:
         loaded = load_with_system_search(libname, found.lib_searched_for)
         if loaded is not None:
             return loaded
+        if libname == "nvvm":
+            # Use cudart as anchor point (libcudart.so.12 is only ~720K, cudart64_12.dll ~560K).
+            loaded_cudart = check_if_already_loaded_from_elsewhere("cudart")
+            if loaded_cudart is not None:
+                found.retry_with_other_abs_path(loaded_cudart.abs_path)
+            else:
+                cudart_abs_path = _load_other_in_subprocess("cudart", found.error_messages)
```

Comment:
I forget the exact rules but CUDA Contexts are not fork safe where if someone has created a CUDA Context before potentially hitting this code under the covers it is problematic.

If we absolutely need to use another process because there's no way for us to resolve the path without possibly loading unwanted symbols, then we need to make sure that we spawn a new process instead of forking a child process.

Reply:
New solution: run_python_code_safely.py, see https://github.com/NVIDIA/cuda-python/pull/604#issuecomment-2852339361